CMakeLists.txt.in

```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
```

Comment:
What does the `.in` file format indicate?

Also, add copyright and license header comment.

Reply:
.in files are input files for cmake.  That's one more of those lovely cmake feaures.
gtest recommend adding the framework like this, 
however, I talked to Charlie and he recommend using another cmake module. I will change this later

Reply:
removed the .in file. using fetch content now